test/gui/shared/scripts/bdd_hooks.py

```diff
@@ -141,3 +145,85 @@ def hook(context):
         raise Exception(
             "Step execution through test middleware failed. Error: " + e.read().decode()
         )
+
+
+def coredump(context):
+    # read coredump location
+    coredumpPath = subprocess.run(
+        ['cat', '/proc/sys/kernel/core_pattern'], stdout=subprocess.PIPE
+    ).stdout.decode('utf-8')
+    coredumpFolderPath = os.path.dirname(coredumpPath)
+    coredumpFilePattern = '^' + os.path.basename(coredumpPath).split('%')[0] + '.*'
+
+    # list items from coredump path
+    ls = subprocess.run(['ls', '-t', coredumpFolderPath], capture_output=True)
+    # get coredump files only
+    coredumps = (
+        subprocess.run(
+            ['grep', coredumpFilePattern],
+            input=ls.stdout,
+            stdout=subprocess.PIPE,
+        )
+        .stdout.decode('utf-8')
+        .split("\n")
+    )
+
+    if coredumps[len(coredumps) - 1] == "":
+        coredumps.pop()
+
+    # write to a file only if there are any coredump files
+    if len(coredumps) > 0:
+        message = "###########################################\n"
+        message += "Scenario: " + context._data["title"] + "\n"
+
+        message += getCoredumpLogs(coredumps, coredumpFolderPath)
+
+        message += "###########################################\n\n"
+
+        print(message)
+        try:
+            coredump_file = os.path.join(
+                os.environ["GUI_TEST_REPORT_DIR"], 'coredumps.md'
+            )
+            open(coredump_file, "a").write(message)
+        except Exception as err:
+            print(err)
+        print("Stacktrace saved.")
+    else:
+        print("No coredump found!")
+
+
+def getCoredumpLogs(coredumps, coredumpFolderPath):
+    message = ""
+    for coredumpFilename in coredumps:
+        coredumpFile = os.path.join(coredumpFolderPath, coredumpFilename)
+        # example coredump file: core-1648445754-1001-11-!drone!src!build_GUI_tests!bin!owncloud
+        appBinary = coredumpFilename.split('!')
+        # remove first element
+        appBinary.pop(0)
+        appBinary = "/" + "/".join(appBinary)
+        print(appBinary)
+
+        message += "-------------------------------------------\n"
```

Comment:
It will be easier to make `message` a list, then call `message.append(...)` for every line. Then, you can `return "\n".join(message)`.